rectools/models/sasrec.py

```diff
@@ -895,21 +906,16 @@ def __init__(
         self.loss = loss
         self.torch_model = torch_model
         self.adam_betas = adam_betas
+        self.gbce_t = gbce_t
+        self.n_item_extra_tokens = n_item_extra_tokens
         self.item_embs: torch.Tensor
 
     def configure_optimizers(self) -> torch.optim.Adam:
         """Choose what optimizers and learning-rate schedulers to use in optimization"""
         optimizer = torch.optim.Adam(self.torch_model.parameters(), lr=self.lr, betas=self.adam_betas)
         return optimizer
 
-    def forward(
-        self,
-        batch: torch.Tensor,
-    ) -> torch.Tensor:
-        """Forward pass. Propagate the batch through torch_model."""
-        return self.torch_model(batch)
-
-    def training_step(self, batch: torch.Tensor, batch_idx: int) -> torch.Tensor:
+    def training_step(self, batch: Dict[str, torch.Tensor]) -> torch.Tensor:
```

Comment:
I don't like that we removed batch_idx. This is a base class. It means that user wants to implement custom logic, he will have to inherit from this class. But if we don't accept batch_idx (or other kwargs here) he cannot do it without breaking everything up :) It's gonna be a mess. Let's please add all of the args that Lightning accepts in all of these kind of methods.

Reply:
fixed